src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt

```diff
@@ -28,594 +28,613 @@ import java.time.OffsetDateTime
+            return createInternalServerError(e.message)
+        }
+        val fieldFilters = fieldController.parseFilters(metaformEntity, fields)
+
+        val replyIdsAndResourceIds = replyController.listIdsAndResourceIds(
+                metaform = metaform,
+                userId = userId,
+                createdBefore = parsedCreatedBefore,
+                createdAfter = parsedCreatedAfter,
+                modifiedBefore = parsedModifiedBefore,
+                modifiedAfter = parsedModifiedAfter,
+                includeRevisions = includeRevisions != null && includeRevisions,
+                fieldFilters = fieldFilters,
+                orderBy = orderBy,
+                latestFirst = latestFirst
+        )
 
-    } catch (e: AuthzException) {
-      return createInternalServerError(e.message!!)
-    }
+        val permittedReplyIds = getPermittedReplies(
+                metaformId = metaformId,
+                replyIdAndResourceIds = replyIdsAndResourceIds,
+                authorizationScope = AuthorizationScope.REPLY_VIEW
+        ).map(ReplyIdAndResourceId::id)
+
+        val resultCount = permittedReplyIds.count().toLong()
+        val resultIds = getReplyIdList(
+                replyIds = permittedReplyIds,
+                firstResult = firstResult,
+                maxResults = maxResults
+        )
 
-    replyController.triggerReplyUpdatedEvent(reply = foundReply)
-    return createNoContent()
-  }
-
-  /**
-   * Returns group permissions based on metaform and reply data
-   *
-   * @param metaformEntity metaform
-   * @param fieldMap field map
-   * @param replyData reply data
-   * @return group permissions based on metaform and reply data
-   */
-  private fun getGroupPermissions(
-    metaformEntity: fi.metatavu.metaform.api.spec.model.Metaform,
-    fieldMap: Map<String, MetaformField>,
-    replyData: Map<String, Any>
-  ): Set<GroupMemberPermission> {
-    val result = mutableSetOf<GroupMemberPermission>()
-
-    replyData.forEach { (fieldName, fieldValue) ->
-      val field = fieldMap[fieldName] ?: return@forEach
-
-      result.addAll(permissionController.getFieldGroupMemberPermissions(
-        field = field,
-        fieldValue = fieldValue
-      ))
-    }
+        val result = resultIds.mapNotNull { replyController.findReplyById(replyId = it) }
+                .map { entity -> replyTranslator.translate(metaformEntity, entity, null) }
+
+        result.forEach { reply ->
+            auditLogEntryController.generateAuditLog(
+                    metaform = metaform,
+                    userId = auditLogUser,
+                    replyId = reply.id!!,
+                    attachmentId = null,
+                    action = null,
+                    type = AuditLogEntryType.LIST_REPLY
+            )
+        }
+        return createOk(result, resultCount)
+    }
+
+    private fun getReplyIdList(replyIds: List<UUID>, firstResult: Int?, maxResults: Int?): List<UUID> {
```

Comment:
Both privates should be located after public methods

Reply:
Moved to the down below ;)